packages/zwave-js/src/lib/node/Node.ts

```diff
@@ -5994,6 +5994,131 @@ ${formatRouteHealthCheckSummary(this.id, otherNode.id, summary)}`,
 		return true;
 	}
 
+	/**
+	 * Returns the current date, time and timezone (or a subset of those) on the node using one or more of the respective CCs.
+	 */
+	public async getDateAndTime(): Promise<object> {
+		const timeParametersAPI = this.commandClasses["Time Parameters"];
+		const timeAPI = this.commandClasses.Time;
+		const clockAPI = this.commandClasses.Clock;
+		const scheduleEntryLockAPI = this.commandClasses["Schedule Entry Lock"];
+
+		// We'll go through the various CCs filling this out and return whatever we've got at the end
+		const response: {
+			hour?: number;
+			minute?: number;
+			second?: number;
+			timezone?: number;
+			weekday?: number;
+			day?: number;
+			month?: number;
+			year?: number;
+		} = {};
+
+		if (
+			timeParametersAPI.isSupported() &&
+			timeParametersAPI.supportsCommand(TimeParametersCommand.Get)
+		) {
+			try {
+				const result = await timeParametersAPI.get();
+				if (result) {
+					// Time Parameters is all UTC per the spec
+					response.hour = result.getUTCHours();
+					response.minute = result.getUTCMinutes();
+					response.second = result.getUTCSeconds();
+					response.timezone = 0;
+					response.weekday = result.getUTCDay();
+					response.day = result.getUTCDate();
+					response.month = result.getUTCMonth();
```

Comment:
Careful, JS months are 0 based for some strange reason.
```suggestion
					response.month = result.getUTCMonth() + 1;
```